revoke/revoke.go

```diff
@@ -60,32 +133,49 @@ func ldapURL(url string) bool {
 //
 //  true, false:  failure to check revocation status causes
 //                  verification to fail
-func revCheck(cert *x509.Certificate) (revoked, ok bool) {
+func (r *Revoke) revCheck(cert *x509.Certificate) (revoked, ok bool) {
+	r.lock.Lock()
+	localCRL := r.localCRL
+	hardFail := r.hardFail
+	r.lock.Unlock()
```

Comment:
There is another lock inside the `certIsRevokedCRL` and it causes double lock, i.e. block the app forever.